lib/userAgent.js

```diff
@@ -65,3 +70,27 @@ function isMobileOS (os) {
   return os.family in MOBILE_OS_FAMILIES
 }
 
+function truncate (userAgentString) {
+  // Completely arbitrary truncation length. This should be a very rare
+  // condition, so we just want something that is long enough to convey
+  // some meaningful information without being too messy.
+  var length = 60
+
+  if (userAgentString.length < length) {
+    return userAgentString
+  }
+
+  if (/.+\(.+\)/.test(userAgentString)) {
+    var openingIndex = userAgentString.indexOf('(')
+    var closingIndex = userAgentString.indexOf(')')
+
+    if (openingIndex < closingIndex && closingIndex < 100) {
+      // If there is a closing parenthesis within a reasonable length,
+      // allow the string to be a bit longer than our arbitrary maximum.
+      length = closingIndex + 1
+    }
+  }
+
+  return userAgentString.substr(0, length)
```

Comment:
One common pattern is to append `"..."` or similar to make it more obvious that the string has been truncated, may be worthwhile to do similarly here.